gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/TypeUtils.java

```diff
@@ -53,6 +54,52 @@ public static boolean isStringType(Type type) {
   }
 
   public static List<TypedExpr> promoteTypeForArithmeticExpressions(List<TypedExpr> expressions) {
```

Comment:
Seems only two inputs are considered for type promotion in arithmetic operations. If so, use left & right as input variables. And revise the implementation accordingly.